BMM.Core/Extensions/ObjectExtensions.cs

```diff
@@ -0,0 +1,16 @@
+using Newtonsoft.Json;
+
+namespace BMM.Core.Extensions
+{
+    public static class ObjectExtensions
+    {
+        /// <summary>
+        /// Simple Copy method using serializing and deserializing JSON.
+        /// </summary>
+        public static T CopyBySerialization<T>(this T objectToCopy)
```

Comment:
are you using this a lot? especially on older android phones this might be expensive

Reply:
It is used to copy single Track object before saving it to persistent memory. As the operation does not touch I/O (everything is in memory), and operates on pretty small single object i don't think it's big overhead.